charts/orchestrator-infra/templates/tests/infra-test.yaml

```
@@ -0,0 +1,83 @@
+{{- if .Values.tests.enabled }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: test-service-account
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook: test
+    helm.sh/hook-delete-policy: before-hook-creation, hook-succeeded, hook-failed
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: test-cluster-role
+  annotations:
+    helm.sh/hook: test
+    helm.sh/hook-delete-policy: before-hook-creation, hook-succeeded, hook-failed
+rules:
+  - apiGroups: ["operators.coreos.com"]
+    resources: ["subscriptions"]
+    verbs: ["list", "get"]
+  - apiGroups: [""]
+    resources: ["namespaces"]
+    verbs: ["list", "get"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: test-cluster-role-binding
+  annotations:
+    helm.sh/hook: test
+    helm.sh/hook-delete-policy: before-hook-creation, hook-succeeded, hook-failed
+subjects:
+  - kind: ServiceAccount
+    name: test-service-account
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: test-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "testing-pod"
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/component: test
+  annotations:
+    helm.sh/hook: test
+spec:
+  serviceAccountName: test-service-account
+  containers:
+    - name: oc-client
+      securityContext:
+        allowPrivilegeEscalation: false
+        readOnlyRootFilesystem: true
+        capabilities:
+          drop: ["ALL"]
+        runAsNonRoot: false
+      resources:
+        requests:
+          cpu: 10m
+          memory: 20Mi
+        limits:
+          cpu: 10m
+          memory: 20Mi
+      image: {{ .Values.tests.image }}
+      command: ["/bin/sh", "-c"]
+      args:
+        - |
+          set -e
+          
+          kubectl get subscription -n {{ .Values.serverlessOperator.subscription.spec.namespace }} || exit 1
```

I guess this should be executed only if `.Values.serverlessOperator.enabled` is `true`? Same point for the subsequent lines..

---

And would it make sense to get the subscription that the Helm Chart created?
```suggestion
          kubectl get subscription "{{ .Values.serverlessOperator.subscription.spec.name }}" -n {{ .Values.serverlessOperator.subscription.spec.namespace }} || exit 1
```